src/main/scala/io/iohk/ethereum/validators/BlockHeaderValidator.scala

```diff
@@ -101,15 +102,21 @@ class BlockHeaderValidatorImpl(blockchainConfig: BlockchainConfig) extends Block
     * Validates [[io.iohk.ethereum.domain.BlockHeader.gasLimit]] follows the restrictions based on its parent gasLimit
     * based on validations stated in section 4.4.4 of http://paper.gavwood.com/
     *
+    * EIP106(https://github.com/ethereum/EIPs/issues/106) adds additional validation of maximum value for gasLimit.
+    *
     * @param blockHeader BlockHeader to validate.
     * @param parentHeader BlockHeader of the parent of the block to validate.
     * @return BlockHeader if valid, an [[HeaderGasLimitError]] otherwise
     */
   private def validateGasLimit(blockHeader: BlockHeader, parentHeader: BlockHeader): Either[BlockHeaderError, BlockHeader] = {
-    val gasLimitDiff = (blockHeader.gasLimit - parentHeader.gasLimit).abs
-    val gasLimitDiffLimit = parentHeader.gasLimit / GasLimitBoundDivisor
-    if(gasLimitDiff < gasLimitDiffLimit && blockHeader.gasLimit >= MinGasLimit) Right(blockHeader)
-    else Left(HeaderGasLimitError)
+    if(blockHeader.gasLimit > MaxGasLimit)
+      Left(HeaderGasLimitError)
+    else {
+      val gasLimitDiff = (blockHeader.gasLimit - parentHeader.gasLimit).abs
+      val gasLimitDiffLimit = parentHeader.gasLimit / GasLimitBoundDivisor
+      if(gasLimitDiff < gasLimitDiffLimit && blockHeader.gasLimit >= MinGasLimit) Right(blockHeader)
```

Comment:
style: space after `if`

Reply:
It would be more concise to just extend this conditional with `MaxGasLimit`

Reply:
I did not want to calculate unnecessary `gasLimitDiff` and `gasLimitDiffLimit`, but this is probably such small thing that for me both versions are ok